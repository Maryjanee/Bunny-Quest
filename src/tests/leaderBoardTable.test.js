import { getByTestId } from "@testing-library/dom";

import createLeaderBoardTable from "../helpers/leaderBoardTable";

describe("LeaderBoardTable Creation ", () => {
  beforeEach(() => {
    let div;
    jest.spyOn(document, "querySelector").mockImplementation((selector) => {
      switch (selector) {
        case "#results":
          div = document.createElement("div");
          div.setAttribute("id", "results");
          div.dataset["testid"] = "results";
          return div;
        default:
          break;
      }
    });
    jest.spyOn(document, "getElementById").mockImplementation((selector) => {
      switch (selector) {
        case "results":
          return div;
        default:
          break;
      }
    });
  });
  const apiData = [
    { user: "Maryjane", score: "30" },
    { user: "Daniel", score: "100" },
    { user: "Steve", score: "0" },
    { user: "Shola", score: "300" },
    { user: "Joe", score: "10" },
    { user: "jane", score: "30" },
    { user: "Mary", score: "50" },
    { user: "Dariel", score: "70" },
    { user: "Afole", score: "0" },
    { user: "Kay", score: "80" },
    { user: "Bisi", score: "40" },
  ];

  it("should create a table element", () => {
    createLeaderBoardTable(apiData);
    const container = document.getElementById("results");
    expect(container).toMatchSnapshot();
  });
  // it('should convert the score property to a number', () => {
  //   expect(typeof returnValue[0].score).not.toBe('string');
  //   expect(typeof returnValue[0].score).toBe('number');
  // });
  // it('should return a sorted array of objects in descending order of scores', () => {
  //   expect(returnValue).toStrictEqual(sortedData);
  // });
});
