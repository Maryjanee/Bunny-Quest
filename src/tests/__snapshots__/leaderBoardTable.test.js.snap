// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoardTable Creation  should create a table element 1`] = `
<div
  data-testid="results"
  id="results"
>
  <table
    class="results-val"
  >
    <thead>
      <tr>
                 
                               
        <th>
          Rank
        </th>
        
                               
        <th>
          Username
        </th>
        
                               
        <th>
          Score
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="score-data"
      >
        
      
        <td>
          1
        </td>
        
      
        <td>
          Maryjane
        </td>
        
                       
        <td>
          30
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          2
        </td>
        
      
        <td>
          Daniel
        </td>
        
                       
        <td>
          100
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          3
        </td>
        
      
        <td>
          Steve
        </td>
        
                       
        <td>
          0
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          4
        </td>
        
      
        <td>
          Shola
        </td>
        
                       
        <td>
          300
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          5
        </td>
        
      
        <td>
          Joe
        </td>
        
                       
        <td>
          10
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          6
        </td>
        
      
        <td>
          jane
        </td>
        
                       
        <td>
          30
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          7
        </td>
        
      
        <td>
          Mary
        </td>
        
                       
        <td>
          50
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          8
        </td>
        
      
        <td>
          Dariel
        </td>
        
                       
        <td>
          70
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          9
        </td>
        
      
        <td>
          Afole
        </td>
        
                       
        <td>
          0
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          10
        </td>
        
      
        <td>
          Kay
        </td>
        
                       
        <td>
          80
        </td>
      </tr>
      <tr
        class="score-data"
      >
        
      
        <td>
          11
        </td>
        
      
        <td>
          Bisi
        </td>
        
                       
        <td>
          40
        </td>
      </tr>
    </tbody>
  </table>
  <p
    class="results-val"
  />
</div>
`;
